aten/src/TH/generic/THTensorRandom.h

```diff
@@ -6,10 +6,10 @@ TH_API void THTensor_(random)(THTensor *self, THGenerator *_generator);
 TH_API void THTensor_(clampedRandom)(THTensor *self, THGenerator *_generator, int64_t min, int64_t max);
 TH_API void THTensor_(cappedRandom)(THTensor *self, THGenerator *_generator, int64_t max);
 TH_API void THTensor_(geometric)(THTensor *self, THGenerator *_generator, double p);
+TH_API void THTensor_(uniform)(THTensor *self, THGenerator *_generator, double a, double b);
```

Comment:
what is the motivation for this change?  It's a bit strange that we have a integer type function taking doubles.

Reply:
Uniform is being used by random so i had to move it out from **#if defined(TH_REAL_IS_FLOAT) || defined(TH_REAL_IS_DOUBLE)** but now im thinking that i should put it into a **#if defined(TH_REAL_IS_FLOAT) || defined(TH_REAL_IS_DOUBLE) || defined(TH_REAL_IS_BOOL)** because it will be generated for all other types now.

Reply:
can you link me the line where it is being used?

Reply:
~/local/pytorch/aten/src/ATen/native/TensorFactories.cpp, rand(IntArrayRef size, Generator* generator, const TensorOptions& options)

line 277